modules/hal_nordic/Kconfig

```diff
@@ -7,6 +7,10 @@ config ZEPHYR_HAL_NORDIC_MODULE
 config HAS_NORDIC_DRIVERS
 	bool
 
+config MPSL
+	bool
+	# Dummy kconfig to allow compiling outside of NCS
```

Comment:
This does not work for all cases downstream, so some other solution needs to be applied.

Reply:
could you propose something then?

Reply:
In https://github.com/nrfconnect/sdk-zephyr/pull/1797#pullrequestreview-2113170934 I proposed to introduce another option that would be selected by `MPSL`. I think this should work.

Reply:
does this work instead: https://github.com/nrfconnect/sdk-nrf/pull/15940/commits/3336767d1a9b8056341821293863ecea8febb7e8

Reply:
@tomchy is there a specific reason why `imply MPSL` was added instead of selecting the closed-source 802154 driver?

Reply:
@anangl do we need to provide `config MSPL` here just to be able
to use
```c
#if defined(CONFIG_MPSL)
```
I think it is not necessary. In downstream CONFIG_MPSL comes from Kconfig, in upstream CONFIG_MPSL is simply undefined.

Reply:
Compliance check fails when there is no `MPSL` option defined but it is referenced.

Reply:
We need to build a global list of PPIs before the preprocessor runs. We have either kconfig, dts or cmake. Since kconfig is a nightmare to follow, and DTS is only for hardware, only cmake is left.

The idea is to have a global list that maps reserved PPIs to their module (e.g. SDC, MPSL, 802) and construct a bitfield out of it that we can pass to nrfx. Each module can add their reserved PPIs from their CMakeLists.txt.

Something like this: https://www.phind.com/search?cache=u25ib9pub0ztwsm50z1h4414

If that sounds good @anangl then I'll prepare a PR.
I still think the [fromlist] should get merged downstream, too bad for the double def of MPSL. We can then revert and use the generic solution after the release.

Reply:
I have a draft PR here: https://github.com/zephyrproject-rtos/zephyr/pull/74711

But moving the logic for the zephyr controller out from the C preprocessor and into cmake seems non-trivial.
I will need help from one of the controller guys @cvinayak @kruithofa